src/rez/tests/test_config.py

```diff
@@ -226,6 +226,22 @@ def _data(self):
         finally:
             os.environ = old_environ
 
+    def test_7(self):
+        """Test path list environment variable with whitespace."""
+        c = Config([self.root_config_file], locked=False)
+
+        # test basic env-var override
+        packages_path = [
+            "c:/foo bar/baz",
+            "c:/foo 2 bar/baz",
+            r"c:\foo bar\baz",
+            "/home/foo bar/baz",
+            r"/home/foo\ bar/baz"
+        ]
+        os.environ["REZ_PACKAGES_PATH"] = os.pathsep.join(packages_path)
```

Comment:
I think it is not a good idea to change an environment variable in the UT as it affects all other tests that are run as part of the same process (i.e when running rez selftests)
This line breaks a test on other unrelated area.
If you really need to, then put the original value back...something along the lines

```
original_packages_path = os.environ["REZ_PACKAGES_PATH"]
os.environ["REZ_PACKAGES_PATH"] = os.pathsep.join(packages_path)
self.assertEqual(c.packages_path, packages_path)
os.environ["REZ_PACKAGES_PATH"] = original_packages_path
```
or if you are going to use it several times create a `@contextmanager` and with `with ...`

Also, I do not understand the original use case of this change.
Why not set the path like `c:/foo;c:/bar/baz` or "/foo:baz/bar` why do you need the space?

Reply:
Agreed on not changing variables that persist across tests, however:

> Why not set the path like c:/foo;c:/bar/baz or "/foo:baz/bar` why do you need the space?

The folder `c:\foo bar\` is different from `c:\foo` and `c:\bar` I think.

Reply:
Would quotes work in this case? ie. "c:\foo bar\"?